app/models/spree/payment.rb

```diff
@@ -3,6 +3,7 @@
 module Spree
   class Payment < ApplicationRecord
     include Spree::Payment::Processing
+    include StripePaymentStatus
```

Comment:
I think this is incorrect OO. The payment abstraction shouldnt have any specific code about the subclasses like Stripe.
The correct structure here is to have a service StripePaymentStatus separate from the payment object that takes a the payment as a parameter.
So instead of payment.stripe_status you can do  StripePaymentStatus.stripe_status(payment) without mixing stripe code into the generic payment class.
Does it make sense? My alternative is quite simple, I think you could switch :+1:

Reply:
This is a good example of why I think concerns are generally concerning.

Reply:
Ha! Yeah I thought this might be controversial. It is incorrect OOP :+1:

I guess I'm posing the question of whether or not a bit of rule-bending is acceptable in order to provide an interface that's nicer for _humans_ in a process that is currently a bit of a nightmare (debuging Stripe issues).

It could just be a service :ok_hand:

Reply:
:-)

Reply:
I think I agree with @luisramos0 here. There's a tiny little bit of knowledge that the `StripePaymentStatus` module has about the Payment class, namely that a Payment has a method `response_code`, but that prevents it from being a truly separate abstraction. Which means it's essentially a way to take these two methods, `stripe_status` and `stripe_captured?` on Payment and tuck them in a separate file. Which is better than just putting them directly in the class, but I think making it a separate service would be better, even if it means that getting the status is a little more awkward to type.